service/create/pki.go

```diff
@@ -0,0 +1,98 @@
+package create
+
+import (
+	"fmt"
+	"strings"
+
+	microerror "github.com/giantswarm/microkit/error"
+
+	"github.com/giantswarm/certctl/service/pki"
+	"github.com/giantswarm/certctl/service/token"
+)
+
+// Creates a PKI backend if one does not exist for the cluster.
+func (s *Service) setupPKIBackend(cert CertificateSpec) error {
+	var err error
+
+	service, err := s.getPKIService()
+	if err != nil {
+		return microerror.MaskAny(err)
+	}
+
+	isValid, err := service.VerifyPKISetup(cert.ClusterID)
+	if err != nil {
+		return microerror.MaskAny(err)
+	}
+	if isValid {
+		s.Config.Logger.Log("debug", fmt.Sprintf("PKI backend already exists for cluster %s", cert.ClusterID))
+		return nil
+	} else {
+		caCommonName := s.Config.getCACommonName(cert)
+
+		// Create PKI backend
+		config := pki.CreateConfig{
+			ClusterID:        cert.ClusterID,
+			CommonName:       caCommonName,
+			AllowedDomains:   getAllowedDomainsForCA(caCommonName, cert),
+			AllowBareDomains: cert.AllowBareDomains,
+			TTL:              s.Config.Viper.GetString(s.Config.Flag.Vault.PKI.CATTL),
+		}
+
+		s.Config.Logger.Log("debug", fmt.Sprintf("Creating PKI backend for cluster %s", cert.ClusterID))
+		return service.Create(config)
```

Comment:
I've opened #11 to cover splitting Create into 2 services. I agree its a cleaner design and I'll also look at the public / private methods problem.